enterprise/backend/test/metabase_enterprise/serialization/api_test.clj

```diff
@@ -0,0 +1,146 @@
+                                                :all-collections false :data-model false :settings true)]
+                    (is (= #{:log :dir :settings}
+                           (with-open [tar (open-tar f)]
+                             (->> (entries tar)
+                                  (map (fn [^TarArchiveEntry e] (file-type (.getName e))))
+                                  set))))))
+
+                (testing "We can export just a single collection"
+                  (let [f (mt/user-http-request :crowberto :post 200 "ee/serialization/export" {}
+                                                :collection (:id coll) :data-model false :settings false)]
+                    (is (= #{:log :dir :dashboard :card :collection}
+                           (with-open [tar (open-tar f)]
+                             (->> (entries tar)
+                                  (map (fn [^TarArchiveEntry e] (file-type (.getName e))))
+                                  set))))))
+
+                (testing "Default export: all-collections, data-model, settings"
+                  (let [f (mt/user-http-request :crowberto :post 200 "ee/serialization/export" {})]
+                    (is (= #{:log :dir :dashboard :card :collection :settings :schema :database}
+                           (with-open [tar (open-tar f)]
+                             (->> (entries tar)
+                                  (map (fn [^TarArchiveEntry e] (file-type (.getName e))))
+                                  set))))))
+
+                (testing "On exception API returns log"
+                  (with-redefs [api.serialization/compress-tgz (fn [& _] (throw (ex-info "Just an error" {})))]
+                    (let [res   (mt/user-http-request :crowberto :post 500 "ee/serialization/export" {}
+                                                      :collection (:id coll) :data-model false :settings false)
+                          lines (str/split-lines (slurp (io/input-stream res)))]
+                      (testing "First three lines for coll+dash+card, and then error during compression"
+                        (is (= #{:collection :dashboard :card}
+                               (log-types (take 3 lines))))
+                        (is (re-find #"ERROR" (nth lines 3))))))))))))
+      (testing "We've left no new files, every request is cleaned up"
+        ;; if this breaks, check if you consumed every response with io/input-stream
+        (is (= known-files
+               (set (.list (io/file api.serialization/parent-dir)))))))))
+
+(deftest export-import-test
+  (testing "Serialization API e2e"
+    (let [known-files (set (.list (io/file api.serialization/parent-dir)))]
+      (premium-features-test/with-premium-features #{:serialization}
+        (binding [api.serialization/*in-tests* true]
+          (testing "POST /api/ee/serialization/export"
+            (mt/with-temp! [Collection coll  {}
```

Comment:
could you explain why we need to use `with-temp!` instead of `with-temp` here?

Reply:
I want to see real requests are passing, since it's possible to return something that will make `user-http-request` pass — it gets raw body you returned and transforms it itself, without invoking all the http server machinery - and does not work in reality. Like, curl saves broken data. So I want `user-real-request` here.